pkg/cmd/docs-issue-generation/requests.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+const (
+	graphQLURL        = "https://api.github.com/graphql"
+	crlJiraBaseUrl    = "https://cockroachlabs.atlassian.net/"
+	jiraRESTURLPart   = "rest/api/3/"
+	jiraDocsUserEmail = "cockroach-jira-docs@cockroachlabs.com"
+)
+
+var (
+	tokenParams      = tokenParameters()
+	githubAuthHeader = fmt.Sprintf("Bearer %s", tokenParams.GitHubToken)
+)
+
+func tokenParameters() apiTokenParameters {
+	const (
+		githubApiTokenEnv = "GITHUB_API_TOKEN"
+		jiraApiTokenEnv   = "JIRA_API_TOKEN"
+	)
+	return apiTokenParameters{
+		GitHubToken: maybeEnv(githubApiTokenEnv, ""),
+		JiraToken:   maybeEnv(jiraApiTokenEnv, ""),
+	}
+}
+
+// queryGraphQL is the function that interfaces directly with the GitHub GraphQL API. Given a query, variables, and
+// token, it will return a struct containing the requested data or an error if one exists.
+func queryGraphQL(query string, queryVariables map[string]interface{}, out interface{}) error {
+	body := map[string]interface{}{
+		"query": query,
+	}
+	if queryVariables != nil {
+		body["variables"] = queryVariables
+	}
+	err := httpRequest(graphQLURL, "POST", "GitHub", nil, body, &out)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	return nil
+}
+
+func queryJiraRESTAPI(
+	apiEndpoint, method string,
+	headers map[string]string,
+	body map[string]interface{},
+	out interface{},
+) error {
+	url := crlJiraBaseUrl + jiraRESTURLPart + apiEndpoint
+	err := httpRequest(url, method, "Jira", headers, body, &out)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	return nil
+}
+
+func httpRequest(
+	url, method, source string, headers map[string]string, body interface{}, out interface{},
```

Comment:
Using a string for the `source` is not ideal as there are only two possible options: `GitHub` and `Jira`. What if I pass in something different? What if someone misspells them or forgets the correct capitalization?

Unfortunately Go doesn't support `enum`s because it's not a real programming language. However, you can kind of implement the same thing using `iota`.

```
type httpReqSource int
const (
    httpSourceGitHub httpReqSource = iota
    httpSourceJira httpReqSource
)
```

Then make `source` an `httpReqSource` instead of a `string`. To be safe, add a check to make sure `source` is either `httpSourceJira` or `httpSourceGitHub` and return an appropriate error if not.